### PR TITLE
Update Windows build script for VS2019

### PR DESCRIPTION
**Pull request by noisyshape**
I updated clean_build.bat with the cmake generator argument for Visual Studio 2019. VS2019 is what Microsoft is pushing now and what people will download if you tell them to download Visual Studio.

I also took the opportunity to move the generator string into its own variable.

### CODE REVIEW

**Comment by p12tic** [comment]
I think we should not break workflow for people who have only 2017 installed. Could the clean_build.bat be modified to check for presence of VS 2019 and if it's not available, build with VS 2017?

**Comment by noisyshape** [comment]
That sounds like a good idea. It looks like VS versions can be identified with the ``VisualStudioVersion`` environmental variable when the developer prompt is being used. I'll do some testing.

**Comment by noisyshape** [comment]
I added a block of code to handle VS2017 and VS2019. If neither is detected, the VS2019 generator is used. The variable can also now be overridden with build_env.bat.

File: clean_build.bat [hunk 1]
```diff
@@ -10,6 +10,18 @@ set B_BONJOUR=C:\Program Files\Bonjour SDK
 set savedir=%cd%
 cd /d %~dp0
 
+REM cmake generator name for the target build system
+if "%VisualStudioVersion%"=="15.0" (
+    set cmake_gen=Visual Studio 15 2017 Win64
+) else if "%VisualStudioVersion%"=="16.0" (
+    set cmake_gen=Visual Studio 16 2019
```
Comment (p12tic) [inline clean_build.bat]: Is `Win64` no longer needed?

Comment (AdrianKoshka) [reply clean_build.bat]: I believe so.

Comment (noisyshape) [reply clean_build.bat]: Barrier doesn't appear to be buildable for Win32. If that's the case, I can normalize the generator strings and specify x64 separately.

Comment (p12tic) [reply clean_build.bat]: Yes, 32-bit build is not currently supported.

**Comment by AdrianKoshka** [comment]
We okay to merge now @p12tic?

**Comment by p12tic** [comment]
Yes.

**Pull request merged**